client/lib/user/support-user-interop.js

```diff
@@ -99,6 +99,55 @@ export const boot = () => {
 
 	const { user, token } = store.get( STORAGE_KEY );
 	debug( 'Booting Calypso with support user', user );
+	store.remove( STORAGE_KEY );
+
+	if ( window && window.localStorage && window.Storage && window.Storage.prototype ) {
+		let memoryStore = {};
+
+		const setItem = Storage.prototype.setItem.bind( window.localStorage );
+		const getItem = Storage.prototype.getItem.bind( window.localStorage );
+		const removeItem = Storage.prototype.removeItem.bind( window.localStorage );
+
+		debug( 'Bypassing localStorage' );
+
+		// The following keys will not be bypassed as
+		// they are safe to share across user sessions.
+		const allowedKeys = [ STORAGE_KEY, 'debug' ];
+
+		Storage.prototype.setItem = ( key, value ) => {
```

Comment:
Since we're overriding the Storage interface, we should be careful and override each interface: https://html.spec.whatwg.org/multipage/webstorage.html#the-storage-interface
```
interface Storage {
  readonly attribute unsigned long length;
  DOMString? key(unsigned long index);
  getter DOMString? getItem(DOMString key);
  setter void setItem(DOMString key, DOMString value);
  deleter void removeItem(DOMString key);
  void clear();
};
```

It looks like we're missing `key(n)` and the length attribute